lib/stm32/common/cordic_common_v1.c

```diff
@@ -0,0 +1,427 @@
+        CORDIC_CSR &= ~CORDIC_CSR_NRES;
+}
+
+/** @brief Set number of CORDIC results to two 32 bit results.
+ *
+ * Use this option for CORDIC operations with two 32 bit results. 
+ * In this case the result ready flag is cleared and a new operation can be started 
+ * as soon as two 32 bit reads from the CORDIC_RDATA register occurred.
+ *
+ */
+void cordic_set_number_of_results_2(void) {
+        CORDIC_CSR |= CORDIC_CSR_NRES;
+}
+
+/** @brief Enable DMA for writes to CORDIC_WDATA
+ *
+ * When enabled, the peripheral will continue to generate DMA requests 
+ * when new arguments can be loaded into the CORDIC_WDATA register.
+ *
+ */
+void cordic_enable_dma_write(void) {
+        CORDIC_CSR |= CORDIC_CSR_DMAWEN;
+}
+
+/** @brief Enable DMA for read from CORDIC_RDATA
+ *
+ * When enabled, the peripheral will continue to generate DMA requests 
+ * when new results can be read from the CORDIC_RDATA register.
+ *
+ */
+void cordic_enable_dma_read(void) {
+        CORDIC_CSR |= CORDIC_CSR_DMAREN;
+}
+
+/** @brief Enable interrupt when result is ready
+ *
+ * When enabled, the peripheral will generate an interrupt 
+ * when the CORDIC_CSR_RRDY flag is set.
+ *
+ */
+void cordic_enable_interrupt(void) {
+        CORDIC_CSR |= CORDIC_CSR_IEN;
+}
+
+/** @brief Set scaling factor for CORDIC operations
+ *
+ * For some operations, the arguments can be multiplied by a factor of 2^-n
+ * to fit in the argument range. The result must then be multiplied by 2^n.
+ * @param[in] n scaling factor
+ *
+ */
+void cordic_set_scaling_factor(uint8_t n) {
+        CORDIC_CSR = (CORDIC_CSR & ~CORDIC_CSR_SCALE_MASK) | (n << CORDIC_CSR_SCALE_SHIFT);
+}
+
+/** @brief Set precision for CORDIC operations
+ *
+ * The speed of CORDIC operations can be increased by lowering the 
+ * number of iterations. This will decrease precision.
+ * @param[in] precision Precision @ref cordic_csr_precision
+ *
+ */
+void cordic_set_precision(uint8_t precision) {
+        CORDIC_CSR = (CORDIC_CSR & ~CORDIC_CSR_PRECISION_MASK) | (precision << CORDIC_CSR_PRECISION_SHIFT);
+}
+
+/** @brief Set CORDIC operation type
+ *
+ * Select what operation the CORDIC peripheral performs.
+ * @param[in] function Function @ref cordic_csr_function
+ *
+ */
+void cordic_set_function(uint8_t function) {
+        CORDIC_CSR = (CORDIC_CSR & ~CORDIC_CSR_FUNC_MASK) | (function << CORDIC_CSR_FUNC_SHIFT);
+}
+
+/** @brief Write single 16 bit argument
+ *
+ * Use this function to set one single 16 bit argument. 
+ * The upper 16 bit of the 32 bit result register 
+ * (that is the second argument) will be set to zero.
+ * @param[in] argument Unsigned int16 argument
+ *
+ */
+void cordic_write_16bit_argument(uint16_t argument) {
+        CORDIC_WDATA = argument;
+}
+
+/** @brief Write two 16 bit arguments
+ *
+ * Use this function to set write 16 bit arguments to the 32 bit CORDIC_WDATA register.
+ * @param[in] argument1 Unsigned int16 argument1
+ * @param[in] argument2 Unsigned int16 argument2
```

Comment:
you don't need the "Unsigned int16" stuff, that's obvious from the signatures...

Reply:
I removed them from all @param but the parameter descriptions became kind of useless. Shall I drop them completely?